bin/pygrb/pycbc_pygrb_pp_workflow

```diff
@@ -16,185 +16,322 @@
-                    level=logging.INFO)
+# =============================================================================
+# Functions
+# =============================================================================
+def make_pygrb_plot(workflow, exec_name, out_dir, # exclude=None, require=None,
+                    ifo=None, inj_set=None, tags=None):
+    """Adds a node for a plot of PyGRB results to the workflow"""
 
-def file_from_workflow(workflow, trig_or_inj, tags=None):
-    """Function to create File object, here used for trigger/injeciton files"""
+    tags = [] if tags is None else tags
 
-    file_out = None
-    if trig_or_inj == 'trig':
-        file_out = wflow.cp.get('workflow', 'trig-file')
-    elif trig_or_inj == 'inj':
-        file_out = wflow.cp.get('workflow', 'inj-file')
-    file_out = _workflow.File.from_path(file_out)
-    if tags is None:
-        file_out.tags = []
+    # Initialize job node with its tags
+    grb_name = workflow.cp.get('workflow', 'trigger-name')
+    extra_tags = ['GRB'+grb_name]
+    if inj_set is not None:
+        #if exec_name not in ['pygrb_plot_snr_timeseries', 'pygrb_plot_chisq_veto',
+        #                     'pygrb_plot_coh_ifosnr', 'pygrb_plot_null_stats',
+        #                     'pygrb_plot_injs_results',
+        #                     'pygrb_plot_stat_distribution', 'pygrb_efficiency']:
+        extra_tags.append(inj_set) # TODO: why is inj_set repeated twice in output files?
+    if ifo:
+        extra_tags.append(ifo)
+    node = PlotExecutable(workflow.cp, exec_name, ifos=workflow.ifos,
+                          out_dir=out_dir,
+                          tags=tags+extra_tags).create_node()
+    # Trigger file
+    if exec_name in ['pygrb_plot_chisq_veto', 'pygrb_plot_coh_ifosnr',
+                     'pygrb_plot_null_stats', 'pygrb_plot_skygrid',
+                     'pygrb_plot_snr_timeseries']:
+        node.add_opt('--trig-file', workflow.cp.get('workflow', 'trig-file'))
+    # Veto directory and category and segments directory
+    node.add_opt('--veto-directory', workflow.cp.get('workflow', 'veto-directory'))
+    node.add_opt('--veto-category', workflow.cp.get('workflow', 'veto-category'))
+    node.add_opt('--segment-dir', workflow.cp.get('workflow', 'segment-dir'))
+    # Other shared tuning values
+    if exec_name not in ['pygrb_plot_skygrid', 'pygrb_plot_coh_ifosnr']:
+        if not (exec_name == 'pygrb_plot_snr_timeseries' and tags[0] != 'reweighted'):
+            for opt in ['chisq-index', 'chisq-nhigh',
+                        'null-snr-threshold', 'snr-threshold', 'newsnr-threshold',
+                        'sngl-snr-threshold', 'null-grad-thresh', 'null-grad-val']:
+                node.add_opt('--'+opt, workflow.cp.get('workflow', opt))
+    # Injection file
+    if inj_set is not None and exec_name != 'pygrb_plot_injs_results':
+        node.add_opt('--inj-file', workflow.cp.get('injections-'+inj_set, 'inj-file'))
```

Comment:
Another FIXME for the future: There's a lot of cases of input files being declared with `add_opt`. In this case pegasus does not know that this is a file, and cannot do any sort of transfer. On a shared filesystem this should always work, but if (for example) you have a large trigger file, and a job might parse through all of it multiple times, it is more efficient to have it copied to a local disk first. Pegasus can handle this for you, but it needs to know these are input files.